libft/sources/map/mapss/ft_mapss_insert.c

```diff
@@ -6,94 +6,108 @@
 /*   By: taretiuk <taretiuk@student.42.fr>          +#+  +:+       +#+        */
 /*                                                +#+#+#+#+#+   +#+           */
 /*   Created: 2024/08/09 13:20:33 by taretiuk          #+#    #+#             */
-/*   Updated: 2024/08/14 21:06:42 by inikulin         ###   ########.fr       */
+/*   Updated: 2024/10/31 15:54:45 by taretiuk         ###   ########.fr       */
 /*                                                                            */
 /* ************************************************************************** */
 
 #include "../../../libft.h"
 #include <stdlib.h>
 
-int	ft_mapss_insert(t_mapss *map, t_dlist *node)
+static int	insert_at_the_end(t_mapss *map, t_dlist *node,
+				t_mapss_entry *node_entry, t_mapss_entry *tail_entry)
 {
-	t_dlist			*current;
-	t_dlist			*tail;
-	t_mapss_entry	*cur_entry;
-	t_mapss_entry	*node_entry;
-	t_mapss_entry	*tail_entry;
+	if (ft_strcmp(node_entry->key, tail_entry->key) > 0)
```

Comment:
Generally opposite design would be better, it makes code less indented hence clearer (also fewer curly braces so less lines for norme)
```
if (ft_strcmp(node_entry->key, tail_entry->key) <= 0)
    return (1);
node->next = NULL;
...
```
Not a blocker, you can mark this as resolved if you don't want to fix this.